### ISSUE TITLE
flash attention 2 - mistral training error - libcudart.so.12

### ISSUE THREAD

**Issue by manishiitg** [skypilot-org/skypilot#2671]
i am using FA2 to fine tune mistral models. 

after release of FA2, i started to get the below error
```
File "/opt/conda/lib/python3.10/site-packages/flash_attn/flash_attn_interface.py", line 8, in
import flash_attn_2_cuda as flash_attn_cuda
ImportError: libcudart.so.12: cannot open shared object file: No such file or directory
Traceback (most recent call last):
```

seems to be CUDA environment issue.

i am using interactive notebooks launched via skypilot on gcp.

not sure the exact fix for this, but this would require changing base image. 

any help in solving this?

below is the yaml on how i am spinning up the notebook

```
# jupyter.yaml

name: jupyter-v

resources:
  accelerators: A100:1
  cloud: gcp
  use_spot: True
  # image_id: skypilot:cuda118-debian-11

file_mounts:
  /sky-notebook:
    source: gs://sky-notebook
    mode: MOUNT

setup: |
  pip install --upgrade pip
  pip install -q notebook
  pip install -q torch torchvision torchaudio
  pip install -q transformers datasets peft trl "safetensors>=0.3.1" --upgrade

  pip install -q trl
  # pip install -q optimum
  pip install -q autoawq
  # pip install -q bitsandbytes
  pip install -q ctranslate2
  pip install -q packaging
  pip install -q accelerate

  git clone https://github.com/OpenAccess-AI-Collective/axolotl.git
  cd axolotl
  pip3 install packaging
  pip3 install -e '.[flash-attn,deepspeed]' -U



run: |
  cd /sky-notebook
  jupyter notebook --port 8888 --allow-root
```

**Comment by manishiitg** [comment]
the same thing works properly when using docker builds
```
docker run --gpus all     -v ~/sky_workdir:/sky_workdir     -v /sky_notebook:/sky_notebook     -v /root/.cache:/root/.cache     winglian/axolotl:main-py3.10-cu118-2.0.1     accelerate launch -m axolotl.cli.train /sky_workdir/qlora.yaml
```

so its a cude env issue

**Comment by Michaelvll** [comment]
Thanks for the question @manishiitg! If it works with the docker image, the problem is likely not related the the nvidia driver but the CUDA runtime installed. It might be because the base environment does not have the latest CUDA runtime.
Does it work by replacing the installation of PyTorch with `conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia` from the PyTorch [doc](https://pytorch.org/get-started/locally/), which will install the CUDA runtime as well?

**Comment by Michaelvll** [comment]
This issue should be fixed now with the #2694 being fixed, i.e. the default image has been upgraded for AWS and GCP to support CUDA 12.2. Please feel free to try this again. I am going to close this issue for now.
